common/remote/rmtspawn.cpp

```diff
@@ -447,6 +447,21 @@ void CRemoteSlave::run(int argc, char * argv[])
                         byte action;
                         msg.read(action);
 
+                        // <= 7.6.xx clients also send:
+                        //      num passwords followed by those pwds
+                        // >= 7.8.0 clients until now do not send password data
+
+                        // so next unsigned could be either num passwords or netaddr
+                        // password info has been set to 0 for many versions
+                        // if next unsigned is not 0 then assumme it is an IP address
+                        // as IP address will surely have higher bits set and be > 0
+
+                        unsigned numPasswds = 0;
+                        size32_t origPos = msg.getPos();
+                        msg.read(numPasswds);
+                        if (numPasswds != 0)
+                            msg.reset(origPos);
```

Comment:
If it's non zero - implying it is actually a 7.6 that has sent passwords, we'd have to cope with deserializing them too. It would hit deserialization problems as it is I think if non-zero.

But I think we can assert that it should always be 0 - @ghalliday ?

Reply:
I just meant we could check for values outside what is reasonable for an IP address.  But if password info sent has been 0 for a long time then I think no need here, just leave it as 0 or not 0 to know.